dnf/cli/commands/history.py

```diff
@@ -276,6 +276,9 @@ def str2transaction_id(s):
                 s = s[4:]
             transaction_id = int(s)
             if transaction_id <= 0:
+                if not self.output.history.last():
+                    logger.critical("Not found given transaction ID")
+                    raise ValueError
```

Comment:
This way of handling the error is wrong (I know it's in parts of the code...), please use this instead of the two log and raise lines (also use proper English and properly describe the error):
```python
raise dnf.cli.CliError(_('Transaction history is empty.'))
```